File: Josephus_Problem.java
import java.util.Scanner;

public class Josephus_Problem
{
	public static void main(String[] args) {
	    	    Scanner sc = new Scanner(System.in);
	    	    int anzahl =0;
	    System.out.println("Sie sitzen im Kreis uns jeder Zweite wird ausgesiebt.");
	    System.out.println("Wer am Schluss übrig bleibt gewinnt.Sie suchen den besten Sitzplatz.");
		System.out.print("Geben Sie die Anzahl der Teilnehmer an: ");
		anzahl = sc.nextInt();
		double a=(Math.log(anzahl) / Math.log(2));
		int b=(int)Math.floor(a);
		int sitzplatz = 2*anzahl-(int) (2*Math.pow(2, b)-1);
		
		System.out.print("Der beste Sitzplatz ist: "+sitzplatz);
		sc.close();
		
		
	}
}

File: josephus_problem.py
def josephus_problem():
    print("Sie sitzen im Kreis und jeder Zweite wird ausgesiebt.")
    print("Wer am Schluss übrig bleibt, gewinnt. Sie suchen den besten Sitzplatz.")
    anzahl = int(input("Geben Sie die Anzahl der Teilnehmer an: "))

    # Berechnung des größten Zweierpotenz kleiner oder gleich anzahl
    potenz = 1 << (anzahl.bit_length() - 1)
    sitzplatz = 2 * (anzahl - potenz) + 1

    print("Der beste Sitzplatz ist:", sitzplatz)

if __name__ == "__main__":
    josephus_problem()